Ginger/GingerCoreNET/Database/NoSqlBase/GingerCassandra.cs

```diff
@@ -37,32 +42,72 @@ public class GingerCassandra : NoSqlBase
         dynamic myclass = null;
         string mUDTName = null;
 
-
-
         public override bool Connect()
         {
             try
             {
-                string queryTimeoutString = "querytimeout=";
+                const string queryTimeoutString = "querytimeout=";
+                const string sslString = "ssl=";
                 int queryTimeout = 20000;//default timeout (20 seconds).
-                if (Db.DatabaseOperations.TNSCalculated.ToLower().Contains(queryTimeoutString.ToLower()))
+                string[] queryArray = Db.DatabaseOperations.TNSCalculated.Split(';');
+                SSLOptions sslOptions = null;
+                for (int i = 1; i < Math.Min(queryArray.Length, 3); i++)
```

Comment:
You are restricting the connection string components to be max 3 but why to have the restriction, no matter how many parameters are passed in future, we just need to loop over them and process them. So please remove the Math.Min(queryArray.Length, 3) and keep just queryArray.Length